guvnor-ala/guvnor-ala-docker-provider/src/main/java/org/guvnor/ala/docker/executor/DockerRuntimeExecExecutor.java

```diff
@@ -51,89 +53,104 @@
         final List<PortBinding> randomPort = new ArrayList<>();
-        final PortBinding randomPortBinding = PortBinding.randomPort( dockerProvider.getHostId() );
+        final PortBinding randomPortBinding = PortBinding.randomPort(dockerProvider.getConfig().getHostIp());
 
-        randomPort.add( randomPortBinding );
-        portBindings.put( runtimeConfig.getPort(), randomPort );
+        randomPort.add(randomPortBinding);
+        portBindings.put(runtimeConfig.getPort(),
+                         randomPort);
 
-        final HostConfig hostConfig = HostConfig.builder().portBindings( portBindings ).build();
+        final HostConfig hostConfig = HostConfig.builder().portBindings(portBindings).build();
 
         final ContainerConfig containerConfig = ContainerConfig.builder()
-                .hostConfig( hostConfig )
-                .image( runtimeConfig.getImage() )
-                .exposedPorts( ports )
+                .hostConfig(hostConfig)
+                .image(runtimeConfig.getImage())
+                .exposedPorts(ports)
                 .build();
 
         final ContainerCreation creation;
         try {
-            creation = docker.getDockerClient( runtimeConfig.getProviderId() ).createContainer( containerConfig );
-            docker.getDockerClient( runtimeConfig.getProviderId() ).startContainer( creation.id() );
-        } catch ( DockerException | InterruptedException ex ) {
-            LOG.error( ex.getMessage(), ex );
-            throw new ProvisioningException( "Error Creating Docker Container with image: " + runtimeConfig.getImage() + "with error: " + ex.getMessage(), ex );
+            creation = docker.getDockerClient(runtimeConfig.getProviderId()).createContainer(containerConfig);
+            docker.getDockerClient(runtimeConfig.getProviderId()).startContainer(creation.id());
+        } catch (DockerException | InterruptedException ex) {
+            LOG.error(ex.getMessage(),
+                      ex);
+            throw new ProvisioningException("Error Creating Docker Container with image: " + runtimeConfig.getImage() + "with error: " + ex.getMessage(),
+                                            ex);
         }
 
         final String id = creation.id();
-        String shortId = id.substring( 0, 12 );
+        String shortId = id.substring(0,
+                                      12);
         String host = "";
         ContainerInfo containerInfo;
         try {
-            containerInfo = docker.getDockerClient( runtimeConfig.getProviderId() ).inspectContainer( id );
-            host = docker.getDockerClient( runtimeConfig.getProviderId() ).getHost();
-        } catch ( DockerException | InterruptedException ex ) {
-            throw new ProvisioningException( "Error Getting Docker Container info: " + id + "with error: " + ex.getMessage(), ex );
+            containerInfo = docker.getDockerClient(runtimeConfig.getProviderId()).inspectContainer(id);
```

Comment:
Same comment above, this code was left this way by the creator.
But I agree, in this case I believe the first docker.getDockerClient is just for "throwing a DockerException" in cases of failures and the containerInfo is really never used.
I'll remove the containerInfo variable to make you happy but preserve the invocation.